src/Rectangle.kt

```diff
@@ -0,0 +1,6 @@
+open class Rectangle(var width: Int, var height: Int) {
+
+    fun setWidth(rectangle: Rectangle, width: Int) {
```

Comment:
Will passing a `Square` here cause unexpected behavior?

Reply:
Oh good catch! Yes, if a `Square` is passed in only its width would be changed. And the definition of a square is that both its sides is equally long, so this is indeed causing unexpected behavior!